doc/neps/nep-0031-uarray.rst

```diff
@@ -94,14 +81,60 @@ vendored into a new namespace within NumPy to give users and downstream
 dependencies access to these overrides.  This vendoring mechanism is similar
 to what SciPy decided to do for making ``scipy.fft`` overridable (see [10]_).
 
+The motivation behind ``uarray`` is manyfold: First, there have been several
+attempts to allow dispatch of parts of the NumPy API, including (most
+prominently), the ``__array_ufunc__`` protocol in NEP-13 [4]_, and the
+``__array_function__`` protocol in NEP-18 [2]_, but this has shown the need
+for further protocols to be developed, including a protocol for coercion (see
+[5]_, [9]_). The reasons these overrides are needed have been extensively
+discussed in the references, and this NEP will not attempt to go into the
+details of why these are needed; but in short: It is necessary for library
+authors to be able to coerce arbitrary objects into arrays of their own types,
+such as CuPy needing to coerce to a CuPy array, for example, instead of
+a NumPy array. In simpler words, one needs things like ``np.asarray(...)`` or
+an alternative to "just work" and return duck-arrays.
 
-Detailed description
---------------------
+Usage and Impact
+----------------
 
-Using overrides
-~~~~~~~~~~~~~~~
+This NEP allows for global and context-local overrides, as well as
+automatic overrides a-la ``__array_function__``.
 
-Here are a few examples of how an end-user would use overrides.
+Here are some use-cases this NEP would enable, besides the 
+first one stated in the motivation section:
+
+The first is allowing alternate dtypes to return their
+respective arrays.
+
+.. code:: python
+
+    # Returns an XND array
+    x = unp.ones((5, 5), dtype=xnd_dtype) # Or torch dtype
+
+The second is allowing overrides for parts of the API.
+This is to allow alternate and/or optimised implementations
+for ``np.linalg``, BLAS, and ``np.random``.
+
+.. code:: python
+
+    import numpy as np
+    import pyfftw # Or mkl_fft
+
+    # Makes pyfftw the default for FFT
+    np.set_global_backend(pyfftw)
+
+    # Uses pyfftw without monkeypatching
+    np.fft.fft(numpy_array)    
+
+    with np.set_backend(pyfftw) # Or mkl_fft, or numpy
+        # Uses the backend you specified
+        np.fft.fft(numpy_array)
```

Comment:
Are you intending to have the uarray functions (e.g. `set_backend`) in both `numpy` as shown here and int `numpy.overridable` as seen in the motivation section?

Reply:
Yes. This one will check for `__ua_domain__ in ["np.fft", "np.linalg", "np.random"]` wheras the one in `numpy.overridable` will check for `__ua_domain__ == "numpy"` or pattern match for `numpy.*`.

The alternative is to have it in each of those submodules.

Reply:
Okay, that seems reasonable.